cpp/oneapi/dal/backend/primitives/ndarray.hpp

```diff
@@ -340,39 +480,74 @@ class ndview : public ndarray_base<axis_count, order> {
         return ndview(new_start_point, new_shape, this->get_strides(), this->data_is_mutable_);
     }
 
+    /// Get 2-dimensional column slice of 2-dimensional view.
+    /// The slice is a view into the original memory block.
+    /// The data is not copied: The sliced view points to the data within the same memory block
+    /// as the original ndview.
+    ///
+    /// @param from_col The starting column index of the data slice within the input view.
+    /// @param to_col   The ending column index of the data slice within the input view.
+    ///
+    /// @return The 2-dimensional view with a data slice containing data columns [from_col, ..., to_col)
+    ///         from the original view.
     template <std::int64_t n = axis_count, typename = std::enable_if_t<n == 2>>
     ndview get_col_slice(std::int64_t from_col, std::int64_t to_col) const {
         return this->t().get_row_slice(from_col, to_col).t();
     }
 
 #ifdef ONEDAL_DATA_PARALLEL
+    /// Creates a multidimensional array with the same shape as the view,
+    /// but having the data allocated on host. The data from the view is copied to the array on host.
+    ///
+    /// @param q    The SYCL* queue object.
+    /// @param deps The vector of events that the operation depends on.
+    ///
+    /// @return The new multidimensional array instance.
     ndarray<T, axis_count, order> to_host(sycl::queue& q, const event_vector& deps = {}) const;
+
+    /// Creates a multidimensional array with the same shape as the view,
+    /// but having the data allocated in USM on device.
+    /// The data from the view is copied to the array on device.
+    ///
+    /// @param q    The SYCL* queue object.
+    /// @param deps The vector of events that the operation depends on.
+    ///
+    /// @return The new multidimensional array instance.
     ndarray<T, axis_count, order> to_device(sycl::queue& q, const event_vector& deps = {}) const;
-#endif
 
-#ifdef ONEDAL_DATA_PARALLEL
+    /// Prefetches the data from high bandwidth memory to local cache.
+    /// Should be submitted ahead the expected computations to have enough time for data transfer.
+    ///
+    /// @param queue The SYCL* queue object.
+    ///
+    /// @return The SYCL* event object indicating the availability of the data in the view for reading
+    ///         and writing.
     sycl::event prefetch(sycl::queue& queue) const {
         return queue.prefetch(data_, this->get_count());
     }
 #endif
 
+    /// Returns the iterator to the first element of the 1-dimensional view.
```

Comment:
nit: This is not returning a generic iterator, it is specifically pointers that are dealt with. Similar comments for the methods below, where `iterator` is mentioned

```suggestion
    /// Returns the pointer to the first element of the 1-dimensional view.
```